src/types/textual.md

```diff
@@ -8,13 +8,13 @@ or 0xE000 to 0x10FFFF range. It is immediate [Undefined Behavior] to create a
 `char` that falls outside this range. A `[char]` is effectively a UCS-4 / UTF-32
 string of length 1.
 
-A value of type `str` is a Unicode string, represented as an array of 8-bit
-unsigned bytes holding a sequence of UTF-8 encoded Unicode code points. Note
-that this is a library-level invariant: for the compiler and core language
-specification, `str` is the same as `[u8]`, but methods working on `str` may
-assume that the data in there is valid UTF-8 and may cause Undefined Behavior
-otherwise. Since `str` is a [dynamically sized type], it can only be
-instantiated through a pointer type, such as `&str`.
+A value of type `str` is represented the same way as `[u8]`, it is a slice of
+8-bit unsigned bytes. However, the Rust standard library makes extra assumptions
+about `str`: methods working on `str` assume and ensure that the data in there
+is valid UTF-8. Calling a `str` method with a non-UTF-8 buffer can cause
+[Undefined Behavior] now or in the future. \
```

Comment:
Is the trailing backslash here on purpose? I can't remember what markdown does to it.

Reply:
It's a newline without starting a new paragraph. I found like the next sentence is somewhat disconnected from the previous ones. You can see that in action by clicking the ["view file" link](https://github.com/rust-lang/reference/blob/ed7ad7290605b96b3c4415cbcd2afb16fce71cb6/src/types/textual.md) in the PR diff view.

I can remove it if you prefer.

Reply:
Ah, as long as it has a purpose, it looks fine to me. I recall there being discussion of changing to one sentence = one line anyways, although it never went anywhere.

Reply:
> I recall there being discussion of changing to one sentence = one line anyways

That's different, that's for just the markdown file.

Reply:
Oh, I see on, uh, the third look at that file. It's subtle, and I don't see a difference in readability either way personally, but if you see it, I'm willing to trust a second set of eyes.